lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1572,16 +1573,24 @@ LogicalResult FIRRTLLowering::visitDecl(WireOp op) {
   // Name attr is required on sv.wire but optional on firrtl.wire.
   auto nameAttr = op.nameAttr() ? op.nameAttr() : builder.getStringAttr("");
 
+  if (!AnnotationSet(op).hasDontTouch())
+    return setLoweringTo<sv::WireOp>(op, resultType, nameAttr);
+  auto moduleName =
+      cast<hw::HWModuleOp>(op->getParentOp()).getNameAttr().getValue().str();
   auto symName = op.nameAttr();
-  if (symName) {
   // Prepend the name of the module to make the symbol name unique in the symbol
   // table, it is already unique in the module. Checking if the name is unique
   // in the SymbolTable is non-trivial.
-    auto moduleName =
-        cast<hw::HWModuleOp>(op->getParentOp()).getNameAttr().getValue().str();
+  if (symName && !symName.getValue().empty())
     symName = builder.getStringAttr("__" + moduleName + "__" +
                                     symName.getValue().str());
```

Comment:
Please use Twine for contatenation here.  Something like this would be more efficient:
  `builder.getStringAttr((Twine("__") + moduleName + "__" + symName.getValue()).str());`

This will be even more efficient when this is available: https://reviews.llvm.org/D103754